### PR TITLE
DOC add type annotations to responsibleai package

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

Adding type annotations to the `responsibleai` package wherever there was anything `Optional`. Would we want these for the required args as well? Also fixing a few smaller formatting inconsistencies and typos in the process.

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [x] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [x] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by imatiach-msft — APPROVED** [review]

File: responsibleai/responsibleai/managers/counterfactual_manager.py [hunk 1]
```diff
@@ -451,36 +452,33 @@ def _add_counterfactual_config(self, new_counterfactual_config):
     def add(self,
             total_CFs,
             method=CounterfactualConstants.RANDOM,
```
Comment (gaugup) [inline responsibleai/responsibleai/managers/counterfactual_manager.py:454]: Can you add type hints for total_CFs (int) and method (str)

File: responsibleai/responsibleai/managers/error_analysis_manager.py [hunk 2]
```diff
@@ -154,8 +155,9 @@ def from_json(json_str):
 class ErrorAnalysisManager(BaseManager):
     """Defines the ErrorAnalysisManager for discovering errors in a model."""
 
-    def __init__(self, model, dataset, target_column, classes=None,
-                 categorical_features=None):
+    def __init__(self, model, dataset, target_column,
```
Comment (gaugup) [inline responsibleai/responsibleai/managers/error_analysis_manager.py:158]: Can you add type hints for model, dataset and target_column?

File: responsibleai/responsibleai/managers/explainer_manager.py [hunk 3]
```diff
@@ -48,7 +49,9 @@ class ExplainerManager(BaseManager):
     """Defines the ExplainerManager for explaining a model."""
 
     def __init__(self, model, initialization_examples, evaluation_examples,
```
Comment (gaugup) [inline responsibleai/responsibleai/managers/explainer_manager.py:51]: Can you add type hints for model, initialization_examples, evaluation_examples and target_column?

File: responsibleai/responsibleai/modelanalysis/model_analysis.py [hunk 4]
```diff
@@ -18,39 +20,16 @@ class ModelAnalysis(object):
...
-        The save method writes the model to file given a parent directory.
-        The load method returns the deserialized model from the same
-        parent directory.
-    :type serializer: object
     """
 
     def __init__(self, model, train, test, target_column,
-                 task_type, categorical_features=None, train_labels=None,
-                 serializer=None,
+                 task_type,
```
Comment (gaugup) [inline responsibleai/responsibleai/modelanalysis/model_analysis.py:26]: same as above?

File: responsibleai/responsibleai/rai_insights/rai_insights.py [hunk 5]
```diff
@@ -50,8 +51,10 @@ class RAIInsights(object):
     """
 
     def __init__(self, model, train, test, target_column,
```
Comment (gaugup) [inline responsibleai/responsibleai/rai_insights/rai_insights.py:53]: same as above

**Review by gaugup — CHANGES_REQUESTED** [review]

**Comment by romanlutz** [comment]
@gaugup if I'm understanding correctly you mean I should add type annotations for every arg, not just the optional ones. Is that correct?

**Comment by gaugup** [comment]
> @gaugup if I'm understanding correctly you mean I should add type annotations for every arg, not just the optional ones. Is that correct?

Type hints are for static checking of the code paths. So a variable being optional or not doesn't matter. We haven't enabled mypy in our builds otherwise you would see these unannotated function variables getting flagged.

**Comment by romanlutz** [comment]
>> @gaugup if I'm understanding correctly you mean I should add type annotations for every arg, not just the optional ones. Is that correct?
> Type hints are for static checking of the code paths. So a variable being optional or not doesn't matter. We haven't enabled mypy in our builds otherwise you would see these unannotated function variables getting flagged.

@gaugup Is that a yes? :-)

**Pull request merged**